backend/src/services/product.service.ts

```diff
@@ -20,6 +20,12 @@ export class ProductService {
     return product.save();
   }
 
+  public async delete(productId: string) {
+    const product = await Product.findByPk(productId);
```

Comment:
The problem is that you never check if the product belongs to the current user or the user isadmin. You shouldn't be able to delete other products

Reply:
I tried to implement a controller wich defines a user as the product owner. Unfortunately I did not find a solution for the error.